nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java

```diff
@@ -0,0 +1,220 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsfc.utils;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.openstack4j.api.Builders;
+import org.openstack4j.model.network.Port;
+import org.openstack4j.model.network.ext.FlowClassifier;
+import org.openstack4j.model.network.ext.PortChain;
+import org.openstack4j.model.network.ext.PortPair;
+import org.openstack4j.model.network.ext.PortPairGroup;
+import org.osc.controller.nsfc.entities.NetworkElementImpl;
+import org.osc.controller.nsfc.entities.PortPairElement;
+import org.osc.controller.nsfc.entities.PortPairGroupElement;
+import org.osc.controller.nsfc.entities.ServiceFunctionChainElement;
+import org.osc.sdk.controller.element.InspectionPortElement;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RedirectionApiUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RedirectionApiUtils.class);
+
+    private OsCalls osCalls;
+
+    public RedirectionApiUtils(OsCalls osCalls) {
+        this.osCalls = osCalls;
+    }
+
+    private NetworkElementImpl fetchNetworkElementWithChildDepends(Port port, String parentId) {
```

Comment:
Can you add comments what this method does? it is not clear from the method name? same for rest of the methods in this class. Also rename the methods with easy to decipher from the class naming.

Reply:
I am removing it and several other methods.